### PR TITLE
Fix missing config copy in Dockerfile

### PR DESCRIPTION
**Pull request by pgraverdy**
Solves #351 

Fix error in Action for docker image creation due to missing conftest.py.

The conftest.py file has been added in Dockerfile, which uses explicit copy of specific files and folders, and not a blanket copy of everything in the build folder except those in .dockerignore. 
While this prevents building a docker image with extra/unwanted files added in the local build folder (on test/dev machines) it also has the potential to create bugs if the Dockerfile has not been properly updated.

@rth @marccarre any recommendation for this ? (do blanket copy anyway as (1) on CICD, no risks or unwanted files in the image and (2) the images built on dev machines are just local anyway).

### CODE REVIEW

**Comment by rth** [comment]
> The conftest.py file has been added in Dockerfile, which uses explicit copy of specific files and folders, and not a blanket copy of everything in the build folder except those in .dockerignore.

I'm not sure I understand. Is this
https://github.com/icubam/icubam/blob/16fb298269d0edd045a03bdcc002e77ba8c90662/conftest.py#L35
the problem? Or just the existence of `conftest.py`? By default that fixture is not used unless you run integration tests: https://github.com/icubam/icubam/blob/master/doc/contributing.md#integration-tests The file needs to be run, but the corresponding fixures are not going to be run unless you pass the `--icubam-config` to pytest.

**Comment by rth** [comment]
Shouldn't some tests fail if tests cannot be run inside Docker though? Do you understand why CI is green on master now?

**Comment by pgraverdy** [comment]
Indeed, the issue was the missing conftest.py in the Dockerfile (was not copied), hence when running the tests inside a container, it was failing. Adding the file inside the Dockerfile has solved the issue AFAICT.

**Comment by rth** [comment]
> Adding the file inside the Dockerfile has solved the issue AFAICT.

OK, great. I think copying files one by one is still fine, because there could be potentially sensitive files in the project dir, particularly when building locally. Ideally if CI fails, it should be easy enough to detect that some file is missing.

File: .dockerignore [hunk 1]
```diff
@@ -2,9 +2,9 @@
...
-deploy_configs
-deploy_configs.tgz
 icubam.egg-info
 docker/configs/certbot
 docker/scripts
-docker/resources
\ No newline at end of file
+docker/resources
+test.db
+icubam.db
```
Comment (rth) [inline .dockerignore:10]: Maybe more generally, `*.db` unless we can't use whildcards in `.dockerignore` ?

Comment (pgraverdy) [reply .dockerignore]: Indeed. replaced with globstar.

**Comment by rth** [comment]
Thanks @pgraverdy !

**Pull request merged**